woodwork/column_accessor.py

```diff
@@ -99,7 +99,10 @@ def init(
             )
 
             if validate:
-                logical_type.validate(self._series)
+                if isinstance(logical_type, (Ordinal, LatLong)):
+                    logical_type.validate(self._series)
+                else:
+                    self._validate_logical_type(logical_type)
```

Comment:
The name `_validate_logical_type` feels a little incorrect for what this method now does, and it adds a bit of ambiguity for what `logical_type.validate` does. I'd suggest either changing the method name to `_validate_dtype` or just move that logic into this else block. Up to you!